library/schema_check.py

```diff
@@ -79,9 +79,9 @@ def check_file_paths(module, *args):
 
 def validate_grp_names(data):
     res_grps = data['resource_groups']
-    res_grp_vars = data['resource_group_vars']
+    res_grp_vars = data['resource_group_vars'] if 'resource_group_vars' in data.keys() else []
     res_grp_names = [ x['resource_group_name'] for x in res_grps ]
-    res_grp_vars = [ x['resource_group_name'] for x in res_grp_vars ]
+    res_grp_vars = [ x['resource_group_name'] for x in res_grp_vars ] if len(res_grp_vars) > 0 else []
```

Comment:
res_grp_vars = [ x['resource_group_name'] if len(res_grp_vars) > 0 else [] for x in res_grp_vars] 

The above seems like it would handle this, no?